### DISCUSSION TITLE
Failed to reach Ring Server

### DISCUSSION THREAD

**Discussion by nocturnal11** [tsightler/ring-mqtt#906] [Support Q&A]
I had the ring-mqtt add-on working for a while and then I added some other add-ons and integrations. Somewhere in there ring-mqtt stopped working with the below logs. I've enabled debugging with *. Is there a way to get more verbose logging?

I've gone to https://api.ring.com/clients_api/profile to validate I can reach it from my network and it show a 401 unauthorized error.

```
Fri, 20 Sep 2024 21:17:01 GMT express:router dispatching GET /
Fri, 20 Sep 2024 21:17:01 GMT express:router query  : /
Fri, 20 Sep 2024 21:17:01 GMT express:router expressInit  : /
Fri, 20 Sep 2024 21:17:01 GMT express:router urlencodedParser  : /
Fri, 20 Sep 2024 21:17:01 GMT body-parser:urlencoded skip empty body
Fri, 20 Sep 2024 21:17:01 GMT send stat "/app/ring-mqtt/web/account.html"
Fri, 20 Sep 2024 21:17:01 GMT send pipe "/app/ring-mqtt/web/account.html"
Fri, 20 Sep 2024 21:17:01 GMT send accept ranges
Fri, 20 Sep 2024 21:17:01 GMT send cache-control public, max-age=0
Fri, 20 Sep 2024 21:17:01 GMT send modified Sat, 17 Aug 2024 01:30:36 GMT
Fri, 20 Sep 2024 21:17:01 GMT send etag W/"92c-1915df53260"
Fri, 20 Sep 2024 21:17:01 GMT send content-type text/html
2024-09-20T21:17:12.611Z ring Failed to reach Ring server at https://api.ring.com/clients_api/profile.  fetch failed.  Trying again in 5 seconds...
2024-09-20T21:17:27.625Z ring Failed to reach Ring server at https://api.ring.com/clients_api/profile.  fetch failed.  Trying again in 5 seconds...
2024-09-20T21:17:42.635Z ring Failed to reach Ring server at https://api.ring.com/clients_api/profile.  fetch failed.  Trying again in 5 seconds...
2024-09-20T21:17:57.645Z ring Failed to reach Ring server at https://api.ring.com/clients_api/profile.  fetch failed.  Trying again in 5 seconds...
2024-09-20T21:18:12.652Z ring Failed to reach Ring server at https://api.ring.com/clients_api/profile.  fetch failed.  Trying again in 5 seconds...
```

**Comment by tsightler** [comment]
What more verbose do you need, the error message is there and quite clear.  Are you running the container version because this looks a lot like a manual install (unsupported) with an outdated version of NodeJS.

**Reply by nocturnal11** [reply]
Thanks for the response, I appreciate you taking time to help out. I'm running HA OS version 2024.9.2 and I followed https://github.com/tsightler/ring-mqtt/wiki/Installation-(Home-Assistant-Addon) for installation.

To me "Failed to reach Ring server" would mean a network error reaching api.ring.com. Assuming that's the error more verbosity would be useful in diagnosing the issue such as IP's or detailed connection logs. I'll go down the network path then and see if I can spin up a Home Assistant in a different network and see if that has any errors then troubleshoot backwards from there.

**Reply by tsightler** [reply]
Understood, but there's nothing I can do in ring-mqtt for this really as fetch is native NodeJS code.  The code just calls the NodeJS fetch API and errors are the same as you would get if you were using a web browser for the URL as fetch is the same API used by JavaScript in web browsers (it's admittedly quite new to NodeJS, only released in 18.x and GA in 20.x).

Failed to connect generally means TCP failed vs, for example, DNS as that would return ENOTFOUND instead.  It could be something like IPv6 being configured but blocked in a firewall or something weird like that, but it's basic networking regardless.  Not much I can do with that.

You can connect to the container while it's running and use curl/wget to troubleshoot it (might have to install these tools with apk, but that's fine since changes to the container are not persistent).

**Reply by nocturnal11** [reply]
Thanks again for the additional explanation; it really helped clarify how the fetch API works within NodeJS and the limitations around logging more verbose network errors from the application side.

After digging further, I found that the issue was indeed network-related, specifically due to a firewall blocking access to the Ring API IP addresses. Your suggestion to troubleshoot networking directly from the container was spot-on, and it led me in the right direction. I appreciate your patience and guidance, which helped me narrow down the problem.

For anyone facing similar issues, checking firewall rules or other network security/routing settings might be worth exploring, especially if the fetch errors are persistent. In this case it was snort blocking IP's in the Firewall (pfsense). I cleared the snort2c table and the connection worked normally.

**Reply by tsightler** [reply]
Thanks for sharing the resolution and sorry for all the typos in that prior mail, I was on a flight and pretty tired so guess I shouldn't answer posts after 14 hours of travel!  I've considered adding some kind of troubleshooting to the web UI, but it's hard to be motivated to add something that is just basic networking stuff.
